src/main/java/co/cask/hydrator/salesforce/plugin/source/streaming/SalesforceReceiver.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.salesforce.plugin.source.streaming;
+
+import co.cask.hydrator.salesforce.authenticator.AuthenticatorCredentials;
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.receiver.Receiver;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation of Spark receiver to receive Salesforce push topic events
+ */
+public class SalesforceReceiver extends Receiver<String> {
+  private static final Logger LOG = LoggerFactory.getLogger(SalesforceReceiver.class);
+  private static final String RECEIVER_THREAD_NAME = "salesforce_streaming_api_listener";
+  // every x seconds thread wakes up and checks if stream is not yet stopped
+  private static final long GET_MESSAGE_TIMEOUT_SECONDS = 2;
+
+  private final AuthenticatorCredentials credentials;
+  private final String topic;
+  private SalesforcePushTopicListener pushTopicListener;
+
+  SalesforceReceiver(AuthenticatorCredentials credentials, String topic) {
+    super(StorageLevel.MEMORY_AND_DISK_2());
+    this.credentials = credentials;
+    this.topic = topic;
+  }
+
+  @Override
+  public void onStart() {
+    pushTopicListener = new SalesforcePushTopicListener(this.credentials, this.topic);
+    pushTopicListener.start();
+
+    new Thread(RECEIVER_THREAD_NAME) {
```

Comment:
never extend Thread class. Use singlethread (or static cached thread pool to make same task execution faster) executor instead. Use executor factory from Apache commons or guava to give thread a name.